app/bundles/LeadBundle/Tests/EventListener/ReportSubscriberTest.php

```diff
@@ -0,0 +1,820 @@
+        $stageModel             = $this->createMock(StageModel::class);
+        $campaignModel          = $this->createMock(CampaignModel::class);
+        $eventCollector         = $this->createMock(EventCollector::class);
+        $companyModel           = $this->createMock(CompanyModel::class);
+        $companyReportDataMock  = $this->createMock(CompanyReportData::class);
+        $fieldsBuilderMock      = $this->createMock(FieldsBuilder::class);
+        $translatorMock         = $this->createMock(Translator::class);
+        $campaignRepositoryMock = $this->createMock(CampaignRepository::class);
+
+        $leadColumns = [
+            'xx.yy' => [
+                'label' => null,
+                'type'  => 'bool',
+                'alias' => 'first',
+            ],
+        ];
+
+        $leadFilters = [
+            'filter' => [
+                'label' => 'second',
+                'type'  => 'text',
+            ],
+        ];
+
+        $companyColumns = [
+            'comp.name' => [
+                'label' => 'company_name',
+                'type'  => 'text',
+            ],
+        ];
+
+        $campaignRepositoryMock = $this->createMock(CampaignRepository::class);
+        $campaignModel->method('getRepository')->willReturn($campaignRepositoryMock);
+
+        if ('companies' != $event) {
+            $fieldsBuilderMock->expects($this->once())
+            ->method('getLeadFieldsColumns')
+            ->with('l.')
+            ->willReturn($leadColumns);
+
+            $fieldsBuilderMock->expects($this->once())
+                ->method('getLeadFilter')
+                ->with('l.', 's.')
+                ->willReturn($leadFilters);
+
+            $companyReportDataMock->expects($this->once())
+            ->method('getCompanyData')
+            ->willReturn($companyColumns);
+        }
+
+        if ('companies' == $event) {
+            $fieldsBuilderMock->expects($this->once())
+            ->method('getCompanyFieldsColumns')
+            ->with('comp.')
+            ->willReturn($companyColumns);
+        }
+
+        $eventCollector->expects($this->any())
+            ->method('getEventsArray')
+            ->willReturn([
+                'decision' => ['email.click' => []],
+            ]);
+
+        $translatorMock->expects($this->any())
+            ->method('hasId')
+            ->willReturn(false);
+
+        $stageModel->expects($this->any())
+            ->method('getUserStages')
+            ->willReturn([
+                'stage' => [
+                    'id'   => '1',
+                    'name' => 'Stage One',
+                ],
+            ]);
+
+        $reportBuilderEvent = new ReportBuilderEvent($translatorMock, $channelListHelperMock, $event, [], $reportHelperMock);
+
+        $reportSubscriber = new ReportSubscriber(
+            $leadModel,
+            $stageModel,
+            $campaignModel,
+            $eventCollector,
+            $companyModel,
```

Comment:
Here you can then use the mocks created in the setUp method.
```suggestion
            $this->companyModel,
```